File: java/FizzBuzzWithMaven/src/main/java/de/welcz/Rule.java
package de.welcz;

class Rule {
    private final int denominator;
    private final String result;

    Rule(int denominator, String result) {
        this.denominator = denominator;
        this.result = result;
    }

    private static boolean isDivisibleBy(int number, int denominator) {
        return number % denominator == 0;
    }

    String getResult() {
        return result;
    }

    boolean appliesTo(int number) {
        return isDivisibleBy(number, denominator);
    }
}

File: typescript/fizzbuzz/src/rule.ts
export class Rule {
    get result(): string {
        return this._result;
    }

    private _denominator: number[];
    private _result: string;

    constructor(result: string, ...denominator: number[]) {
        this._denominator = denominator;
        this._result = result;
    }

    appliesTo(number: number): boolean {
        return this._denominator.every(d => number % d == 0);
    }
}
